lib/Listener/SpeechToText/SpeechToTextResultListener.php

```diff
@@ -49,20 +55,69 @@ public function handle(Event $event): void {
 		if ($event instanceof TranscriptionSuccessfulEvent) {
 			$transcript = $event->getTranscript();
 			$userId = $event->getUserId();
+			$file = $event->getFile();
 
+			$tasks = $this->taskMapper->getTasksByOcpTaskIdAndCategory($file->getId(), Application::TASK_GATEGORY_SPEECH_TO_TEXT);
+
+			// Find a matching etag:
+			$etag = $file->getEtag();
+			$assistantTask = null;
+			foreach ($tasks as $task) {
+				$taskEtag = $task->getInputsAsArray()['eTag'];
+				if ($taskEtag === $etag) {
+					$assistantTask = $task;
+					break;
+				}
+			}
```

Comment:
Why would there be multiple tasks with the same ocp_task_id?

Reply:
That's an issue specific to the speech-to-text API; it doesn't have task ids so the only way to resolve the notification event back to the assistant task is by using the file id which is substituted for the ocpTaskId in this case. Then it's possible that you could have multiple tasks for the same file id (and possibly multiple different file versions)..?